src/vorta/keyring/keyring_secretstorage.py

```diff
@@ -0,0 +1,22 @@
+from .keyring_backend import KeyringBackend
+import secretstorage
+
+
+class VortaSecretStorageKeyring(KeyringBackend):
+    """A wrapper for the secretstorage package to support the custom keyring backend"""
+
+    def set_password(self, service, repo_url, password):
+        connection = secretstorage.dbus_init()
+        collection = secretstorage.get_default_collection(connection)
+        attributes = {'application': service, 'repo_url': repo_url}
+        collection.create_item(f"vorta-{repo_url}", attributes, password.encode(), replace=True)
+
+    def get_password(self, service, repo_url):
+        connection = secretstorage.dbus_init()
+        collection = secretstorage.get_default_collection(connection)
+        attributes = {'application': service, 'repo_url': repo_url}
+        try:
+            items = collection.search_items(attributes)
+            return next(items).get_secret().decode()
+        except Exception:
+            return None
```

Comment:
diaper pattern detected :)

Reply:
I don't know what you mean with that^^

I included it, because @m3nu did something similar in keyring_db.py and I hoped it was the reason my implementation didn't work (now we know it is a issue with dbus).

Reply:
He means the `except Exception`. It should be more specific. "It's [called](https://mike.pirnat.com/2009/05/09/the-diaper-pattern-stinks/) a Diaper because it catches all the shit."